### PR TITLE
Ensure unicode strings for Python 2 / 3 compat

### PR DESCRIPTION
**Pull request by catleeball**
Made several changes to `credentials_lib.py` for Python 2/3 compatibility. Specifically, Running pre-release tests, we found some of these methods would fail when gsutil had no configs available.

### CODE REVIEW

File: apitools/base/py/credentials_lib.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 # limitations under the License.
 
 """Common credentials classes and constructors."""
+from __future__ import unicode_literals
```
Comment (houglum) [inline apitools/base/py/credentials_lib.py]: I'd rather not import this in just one file; I don't think most of the other files in this directory use this import.  (If we're going to use unicode literals, we should do it across the entire library.)

File: apitools/base/py/credentials_lib.py [hunk 2]
```diff
@@ -352,7 +356,7 @@ def _ScopesFromMetadataServer(self, scopes):
     def GetServiceAccount(self, account):
         relative_url = 'instance/service-accounts'
         response = _GceMetadataRequest(relative_url)
-        response_lines = [line.rstrip('/\n\r')
+        response_lines = [six.ensure_text(line).rstrip('/\n\r')
```
Comment (houglum) [inline apitools/base/py/credentials_lib.py]: If we're going to remove the `unicode_literals` import, we should make sure we're stripping off unicode here (i.e. adding the `u` prefix: `rstrip(u'/\n\r')`)

Comment (catleeball) [reply apitools/base/py/credentials_lib.py]: Makes sense to me! Made both changes. :)

**Comment by catleeball** [comment]
@houglum Thanks for the approval! I don't have write access to the apitools repo, so feel free to merge this when you have a moment :slightly_smiling_face:

**Pull request merged**